BrowserKit/Sources/SiteImageView/ImageProcessing/ImageHandler.swift

```diff
@@ -117,7 +117,13 @@ class DefaultImageHandler: ImageHandler {
             await imageCache.cacheImage(image: image, cacheKey: imageModel.cacheKey, type: imageModel.imageType)
             return image
         } catch {
-            return await fallbackToLetterFavicon(imageModel: imageModel)
+            let isServerError = ServerErrorHelper.extractServerError(error)
```

Comment:
Unit tests for this as well and the `fallbackToLetterFavicon` code would be awesome 🙏